zuul-core/src/main/java/com/netflix/zuul/netty/insights/PassportLoggingHandler.java

```diff
@@ -99,21 +99,31 @@ private void logPassport(Channel channel)
                     + ", toplevelid = " + topLevelRequestId
                     + ", req = " + request.getInfoForLogging()
                     + ", passport = " + String.valueOf(passport));
-        }
 
-        // Some logging of session states if certain criteria match:
-        if (LOG.isInfoEnabled()) {
-            if (passport.wasProxyAttempt()) {
+            StartAndEnd inReqToOutResp = passport.findFirstStartAndLastEndStates(PassportState.IN_REQ_HEADERS_RECEIVED, PassportState.OUT_REQ_LAST_CONTENT_SENT);
+            if (passport.calculateTimeBetween(inReqToOutResp) > WARN_REQ_PROCESSING_TIME_NS.get()) {
+                LOG.debug("Request processing took longer than threshold! toplevelid = " + topLevelRequestId + ", "
+                        + ChannelUtils.channelInfoForLogging(channel));
+            }
 
-                if (passport.findStateBackwards(PassportState.OUT_RESP_LAST_CONTENT_SENDING) == null) {
-                    incompleteProxySessionCounter.increment();
-                    LOG.info("Incorrect final state! toplevelid = " + topLevelRequestId + ", " + ChannelUtils.channelInfoForLogging(channel));
-                }
+            StartAndEnd inRespToOutResp = passport.findLastStartAndFirstEndStates(PassportState.IN_RESP_HEADERS_RECEIVED, PassportState.OUT_RESP_LAST_CONTENT_SENT);
+            if (passport.calculateTimeBetween(inRespToOutResp) > WARN_RESP_PROCESSING_TIME_NS.get()) {
+                LOG.debug("Response processing took longer than threshold! toplevelid = " + topLevelRequestId + ", "
+                        + ChannelUtils.channelInfoForLogging(channel));
             }
+        }
 
-            if (! passport.wasProxyAttempt()) {
-                if (ctx != null && !isHealthcheckRequest(request)) {
-                    // Why did we fail to attempt to proxy this request?
+        if (passport.wasProxyAttempt()) {
+            if (passport.findStateBackwards(PassportState.OUT_RESP_LAST_CONTENT_SENDING) == null) {
+                incompleteProxySessionCounter.increment();
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Incorrect final state! toplevelid = " + topLevelRequestId + ", " + ChannelUtils.channelInfoForLogging(channel));
+                }
+            }
+        } else {
+            if (ctx != null && !isHealthcheckRequest(request)) {
+                // Why did we fail to attempt to proxy this request?
+                if (LOG.isDebugEnabled()) {
```

Comment:
This is redundant, and no longer necessary with the log4j implementation of slf4j. 
It'd actually make sense to clean up the conditionals here.

Reply:
The input to LOG.debug below is evaluated before the logger method itself does the log level check. The advice I received is to wrap the call with this conditional if we're constructing a log message dynamically.

Reply:
Hmm, I left a separate comment around string concat but there have been optimizations there. 
It's difficult to discuss this without an allocation profile imo.